src/common/com/intellij/plugins/haxe/model/type/ResultHolder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2015 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.model.type;
+
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class ResultHolder {
+  static public ResultHolder[] EMPTY = new ResultHolder[0];
+
+  @NotNull
+  private SpecificTypeReference type;
+
+  private boolean canMutate = true;
+  private int mutationCount = 0;
+
+  public ResultHolder(@NotNull SpecificTypeReference type) {
+    this.type = type;
+  }
+
+  @NotNull
+  public SpecificTypeReference getType() {
+    return type;
+  }
+
+  public SpecificFunctionReference getFunctionType() {
+    return (SpecificFunctionReference)type;
```

Comment:
Sure. I was calling isFunctionType (instead of returning null and checking for null) before calling this so it was not a problem. But I can mark the function as Nullable and check it inside and return null in the case it is not a function type.